File: solutions/php/0283-move-zeroes.php
<?php

class Solution
{
    /**
     * @param Integer[] $nums
     * @return NULL
     */
    public function moveZeroes(array &$nums): void
    {
        foreach ($nums as $key => $value) {
            if ($value === 0) {
                unset($nums[$key]);
                $nums[] = 0;
            }
        }
    }

    // Another (less efficient) solution with a custom sort function
    public function moveZeroes2(array &$nums): void
    {
        usort($nums, function (int $a, int $b): int {
            if ($a !== 0 && $b !== 0) {
                return 0;
            }

            if ($a === $b) {
                return 0;
            }

            return $a === 0 ? 1 : -1;
        });
    }
}

File: solutions/go/0283-move-zeroes.go
package main

// Another (less efficient) solution
func moveZeroes2(nums []int) {
	var i int
	var zeroes int

	for i < len(nums)-zeroes {
		if nums[i] != 0 {
			i++
			continue
		}

		zeroes++
		for j := i + 1; j < len(nums); j++ {
			nums[j-1], nums[j] = nums[j], nums[j-1]
		}
	}
}

func moveZeroes(nums []int) {
	var j int

	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			continue
		}

		nums[i], nums[j] = nums[j], nums[i]
		j++
	}
}
